maven-core/src/main/java/org/apache/maven/ReactorReader.java

```diff
@@ -263,15 +271,28 @@ private boolean isPackagedArtifactUpToDate( MavenProject project, File packagedA
             while ( iterator.hasNext() )
             {
                 Path outputFile = iterator.next();
+
+                if ( Files.isDirectory(  outputFile ) )
+                {
+                    continue;
+                }
+
                 long outputFileLastModified = Files.getLastModifiedTime( outputFile ).toMillis();
                 if ( outputFileLastModified > artifactLastModified )
                 {
-                    LOGGER.warn(
-                            "Packaged artifact for {} is not up-to-date compared to the build output directory; "
-                          + "file {} is more recent than {}.",
-                            project.getArtifactId(),
-                            relativizeOutputFile( outputFile ), relativizeOutputFile( packagedArtifactFile.toPath() )
-                    );
+                    File alternative = determineLooseDirectoryForArtifact( project, artifact );
+                    if ( alternative != null )
+                    {
+                        LOGGER.warn( "File {} is more recent than the packaged artifact for {}; using {} instead",
```

Comment:
Please put args to logging in single quotes if they are non-numbers or collections.

Reply:
I wouldn't mind, but is that some new kind of rule? I see many other examples that do not follow that rule, like

> Compiling 54 source files to /Users/maarten/Code/open-source/maven/maven/maven-model/target/classes

or

> skip non existing resourceDirectory /Users/maarten/Code/open-source/maven/maven/maven-xml/src/main/resources

Both paths are non-numbers or collections, but they aren't between single quotes either.

Reply:
It is inconsitent throughout the codebase. This needs to be addressed. There is no ticket yet for this.